src/ussd_popi_user_data.js

```diff
@@ -420,32 +420,112 @@ go.app = function() {
+
+                if (identities === null) {
+                    return self.states.create('state_create_msisdn_change');
+                }
+
+                var opted_out_on_operator = false;
+                var msisdn_available = true;
+                for(i=0; i < identities.length; i++) {
+                    // Msisdn already on this operator but inactive
+                    if (identities[i].id === self.im.user.answers.operator.id) {
+                      if (identities[i].details.addresses.msisdn[new_msisdn].optedout ||
+                              identities[i].details.addresses.msisdn[new_msisdn].inactive)
+                        opted_out_on_operator = true;
+                    }
+                    // Msisdn active on any other than this operator
+                    if (!identities[i].details.addresses.msisdn[new_msisdn].optedout &&
+                            !identities[i].details.addresses.msisdn[new_msisdn].inactive) {
+                        if (identities[i].id !== self.im.user.answers.operator.id) {
+                            msisdn_available = false;
+                            break;
+                        }
+                    }
                 }
+
+                if (!msisdn_available) {
+                    return self.states.create('state_msisdn_change_fail');
+                }
+
+                if (opted_out_on_operator) {
+                    return self.states.create('state_opt_in');
+                }
+
+                return self.states.create('state_create_msisdn_change');
+            });
+        });
+
+        self.add('state_msisdn_change_fail', function(name) {
+            return new EndState(name, {
+                text: $("Sorry, the number you are trying to move to already has an active registration. To manage that registration, please redial from that number."),
+                next: 'state_start'
+            });
+        });
+
+        self.add('state_create_msisdn_change', function(name) {
+            var change_info = {
+                "registrant_id": self.im.user.answers.operator.id,
+                "action": "momconnect_change_msisdn",
+                "data": {
+                    "msisdn": self.im.user.answers.new_msisdn
+                }
+            };
+
+            self.im.user.set_answer("operator_msisdn", self.im.user.answers.new_msisdn);
+            return hub.create_change(change_info)
+            .then(function() {
+                return self.states.create('state_updated');
+            });
+        });
+
+        self.states.add("state_opt_in", function(name) {
+            var optin_info = {
+                "identity": self.im.user.answers.operator.id,
+                "address_type": "msisdn",
+                "address": self.im.user.answers.new_msisdn,
+                "request_source": self.im.config.name || "ussd_popi_user_data"
+            };
+            return is
+            .optin(optin_info)
```

Comment:
Hmmm good point. Investigation says not (https://github.com/praekelt/seed-identity-store/blob/develop/identities/models.py#L126).
I'll get it to create a change as well so that it gets set as the default.